### PR TITLE
New animations on FireMage Spells and prefabs

### PR DESCRIPTION
**Pull request by Myrilandel**
## New Features
**Club**
- The club as his first prefabs/deco finished

**SolarBall**
- Script associated with the SolarBurn prefab.
- Allows the prefab to launch itself (extends LinearProjectile).
- Launches animations of the prefab.

## Enhancement
**AutoAttackFireMage**
- Now displays a Shield animation with a new prefab.

**IgniteStatus**
- Now apply a prefab with animations on targets.
- Now have a method ExplodeIgnite that can be activated by ConflagrationSpell.

**LinearProjectile**
- Now has a method AdditionalEffects() to give the ability to daughter classes to Apply effects (such as animations or area of effects) no matter if they it an entityliving or not. 
- Now has an override of method LaunchProjectile with a target in argument for special linearProjectiles that do not follox the classic way From hand to target.

**PosHelper**
- Now has a method GetRightHandTransformOfPlayer that returns the Transform of the right hand of the player in a static context. Useful to instanciate the prefabs associated to spells.

## Bug Fixes
**FireballSpell**
- Now correctly get the hand of the player to be instanciated.

**AutoAttackFireMage**
- Now correctly get the hand of the player to be instanciated.

## Minor Fixes
**GameObjectDetector**
- Elements around the player (shields for example) were sometimes detected to be transparent. It is now fixed.

**PassiveFireMage**
- Now correctly Start with a numberOfIgnite = 0.

## KB Tasks & PlantUML
- [KB Task #94](https://velandelstudio.com/kanboard/?controller=TaskViewController&action=show&task_id=94&project_id=3)
- [KB Task #105](https://velandelstudio.com/kanboard/?controller=TaskViewController&action=show&task_id=105&project_id=3)
- [KB Task #117](https://velandelstudio.com/kanboard/?controller=TaskViewController&action=show&task_id=117&project_id=3)
- [KB Task #124](https://velandelstudio.com/kanboard/?controller=TaskViewController&action=show&task_id=124&project_id=3)
- [KB Task #128](https://velandelstudio.com/kanboard/?controller=TaskViewController&action=show&task_id=128&project_id=3)
- [KB Task #130](https://velandelstudio.com/kanboard/?controller=TaskViewController&action=show&task_id=130&project_id=3)
_Weird indentation on git but the code on visual studio seems to be OK_

### CODE REVIEW

**Comment by Myrilandel** [comment]
Happy Review les amis ! 
La PR est assez conséquente, elle balaye pas mal de scripts avec des retouches par-ci par là. N'hésitez pas à me demander des infos, si ce n'est pas clair !
EM

File: Assets/Scripts/Classes/FireMage/ScriptsOnPrefabs/SolarBall.cs [hunk 1]
```diff
@@ -4,12 +4,53 @@
...
+			}
+		}
+		
+		for(int i = 0; i < 3; i ++)
+		{
+			Vector2 pointInCircle = Random.insideUnitCircle.normalized * 3;
+			Vector3 v = new Vector3(transform.position.x + pointInCircle.x, transform.position.y, transform.position.z + pointInCircle.y);
+			Instantiate(_particlesOnFloor.gameObject, v, _particlesOnFloor.rotation);
+		}
+	}
```
Comment (Veltouille) [inline Assets/Scripts/Classes/FireMage/ScriptsOnPrefabs/SolarBall.cs:55]: Bonne idée d'avoir utilisé une nouvelle méthode pour ce qui ne concerne pas directement des effets sur les entities.

File: Assets/Scripts/Mechanisms/ProjectilesMecanism/LinearProjectile.cs [hunk 2]
```diff
@@ -116,4 +139,6 @@ protected float CalculTimeOfFly (float speed, float distance)
     {
         return distance / speed;
     }
+	
+	protected virtual void AdditionalEffects(){}
```
Comment (Veltouille) [inline Assets/Scripts/Mechanisms/ProjectilesMecanism/LinearProjectile.cs:143]: Je pense qu'il faudrait ajouter cette méthode directement dans l'Iprojectile.

**Review by Veltouille — APPROVED** [review]
Franchement encore nickel !

Il y avait beaucoup de petites modifs mais rien de très difficile ! Regarde en particulier pour l'ajout de additionalEffect pour l'interface ! Sinon c'est parfait !!

Je valide !

File: Assets/Scripts/Mechanisms/ProjectilesMecanism/LinearProjectile.cs (hunk 2 above)
Comment (Myrilandel) [reply Assets/Scripts/Mechanisms/ProjectilesMecanism/LinearProjectile.cs:143]: Oui on pourra

**Comment by Myrilandel** [comment]
Merci pour ton commentaire, on ajoutera ca dans une prochaine PR ! je merge

**Pull request merged**